common/src/main/cpp/extensions/openxr_fb_composition_layer_secure_content_extension_wrapper.cpp

```diff
@@ -0,0 +1,135 @@
+
+	request_extensions[XR_FB_COMPOSITION_LAYER_SECURE_CONTENT_EXTENSION_NAME] = &fb_composition_layer_secure_content;
+	singleton = this;
+}
+
+OpenXRFbCompositionLayerSecureContentExtensionWrapper::~OpenXRFbCompositionLayerSecureContentExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbCompositionLayerSecureContentExtensionWrapper::_bind_methods() {
+}
+
+void OpenXRFbCompositionLayerSecureContentExtensionWrapper::cleanup() {
+	fb_composition_layer_secure_content = false;
+}
+
+Dictionary OpenXRFbCompositionLayerSecureContentExtensionWrapper::_get_requested_extensions() {
+	Dictionary result;
+	for (auto ext : request_extensions) {
+		uint64_t value = reinterpret_cast<uint64_t>(ext.value);
+		result[ext.key] = (Variant)value;
+	}
+	return result;
+}
+
+uint64_t OpenXRFbCompositionLayerSecureContentExtensionWrapper::_set_viewport_composition_layer_and_get_next_pointer(const void *p_layer, const Dictionary &p_property_values, void *p_next_pointer) {
+	if (!fb_composition_layer_secure_content) {
+		return reinterpret_cast<uint64_t>(p_next_pointer);
+	}
+
+	const XrCompositionLayerBaseHeader *layer = reinterpret_cast<const XrCompositionLayerBaseHeader *>(p_layer);
+
+	if (!layer_structs.has(layer)) {
+		layer_structs[layer] = {
+			XR_TYPE_COMPOSITION_LAYER_SECURE_CONTENT_FB, // type
+			p_next_pointer, // next
+			0, // flags
+		};
+	}
+
+	XrCompositionLayerSecureContentFB *secure_content = layer_structs.getptr(layer);
+
+	switch ((ExternalOutput)(int)p_property_values.get(EXTERNAL_OUTPUT_PROPERTY_NAME, EXTERNAL_OUTPUT_DISPLAY)) {
+		case EXTERNAL_OUTPUT_DISPLAY: {
+			secure_content->flags = 0;
+		} break;
+		case EXTERNAL_OUTPUT_EXCLUDE: {
+			secure_content->flags = XR_COMPOSITION_LAYER_SECURE_CONTENT_EXCLUDE_LAYER_BIT_FB;
+		} break;
+		case EXTERNAL_OUTPUT_REPLACE: {
+			secure_content->flags = XR_COMPOSITION_LAYER_SECURE_CONTENT_REPLACE_LAYER_BIT_FB;
+		} break;
+	};
+
+	return reinterpret_cast<uint64_t>(secure_content);
+}
+
+TypedArray<Dictionary> OpenXRFbCompositionLayerSecureContentExtensionWrapper::_get_viewport_composition_layer_extension_properties() {
+	TypedArray<Dictionary> properties;
+
+	{
+		Dictionary external_output;
+		external_output["name"] = EXTERNAL_OUTPUT_PROPERTY_NAME;
+		external_output["type"] = Variant::INT;
+		external_output["hint"] = PROPERTY_HINT_ENUM;
+		external_output["hint_string"] = "Display,Exclude,Replace";
+		properties.push_back(external_output);
+	}
+
+	return properties;
+}
+
+Dictionary OpenXRFbCompositionLayerSecureContentExtensionWrapper::_get_viewport_composition_layer_extension_property_defaults() {
+	Dictionary defaults;
+	defaults[EXTERNAL_OUTPUT_PROPERTY_NAME] = (int)EXTERNAL_OUTPUT_DISPLAY;
+	return defaults;
+}
+
+void OpenXRFbCompositionLayerSecureContentExtensionWrapper::_on_viewport_composition_layer_destroyed(const void *p_layer) {
+	const XrCompositionLayerBaseHeader *layer = reinterpret_cast<const XrCompositionLayerBaseHeader *>(p_layer);
```

Comment:
Similar comment here about checking the extension is enabled. 
Here it's not as important since we're only trying to remove an inexistent value from the map, but it guards against future logic additions that may extend the current logic, and establish a pattern for other composition layer additions to follow.

Reply:
Ok, this one is only for runtime, so we can add the check here :-)

Reply:
Fixed in latest push